src/palette.ml

```diff
@@ -120,3 +120,18 @@ let index_to_rgb (palette : t) (index : int) : int32 =
 
 let to_list (palette : t) : int list =
     List.map Int32.to_int (Array.to_list palette)
+
+let circle_palette (pal : t) (offset : int) : t =
+  let size = Array.length pal in
+  Array.init size (fun index ->
+    let new_index = (index + offset) mod size in
+    pal.(new_index)
+  )
+
+let updated_entry (pal : t) (index : int) (new_colors : int32) : t =
+  let new_pal = Array.copy pal in
+  if index < 0 || index >= Array.length new_pal then
```

Comment:
I'd generally opt to do this test first, so that you don't duplicate the array unnecessarily. What you've done works fine, but will do that copy pointlessly if the index isn't valid.